source/MaterialXCore/Document.cpp

```diff
@@ -401,6 +402,32 @@ void Document::invalidateCache()
     _cache->valid = false;
 }
 
+void Document::normalizeValueStrings(unsigned int numericPrecision)
+{
+    // For now this will only normalize numeric strings
+    StringSet numericType = { "float", "vector2", "vector3", "vector4", "color3", "color4" };
+    for (ElementPtr elem : traverseTree())
+    {
+        ValueElementPtr valueElem = elem->asA<ValueElement>();
+        if (!valueElem || numericType.end() == numericType.find(valueElem->getType()))
```

Comment:
I think this check would be better done inside of `normalizeNumericString()` itself and just pass along the type name of the value being normalized.  Otherwise other callers of this function will likely have to replicate this logic elsewhere.